### PR TITLE
optimize/functions: fix BrownBadlyScaled for BFGS and LBFGS tests

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

Updates #1218.

### CODE REVIEW

File: optimize/functions/functions.go [hunk 1]
```diff
@@ -553,7 +553,7 @@ func (BrownBadlyScaled) Grad(grad, x []float64) {
 
 	f1 := x[0] - 1e6
 	f2 := x[1] - 2e-6
-	f3 := x[0]*x[1] - 2
+	f3 := float64(x[0]*x[1]) - 2
```
Comment (sbinet) [inline optimize/functions/functions.go]: perhaps add a comment about that? (or just mention `#1218`?)

Comment (kortschak) [reply optimize/functions/functions.go]: I've put a tag comment that should be enough for people to look at the commit message for this line.

**Comment by kortschak** [comment]
PTAL

**Comment by kortschak** [comment]
PTAL (sorry - fixed commit message and so dismissed review).

**Pull request merged**